### ISSUE TITLE
Does this Plugin work with enhanced input?

### ISSUE THREAD

**Issue by luciddream-tsin** [JaydenMaalouf/JoystickPlugin#48]
axis mapping is abandoned  in UE5.2， does this plugin support enhanced input?

**Comment by JaydenMaalouf** [comment]
> axis mapping is abandoned  in UE5.2， does this plugin support enhanced input?

It doesn't yet but I'll look at migrating it to it because it handles some of the remapping stuff quite nicely

Thanks for pointing this out!

**Comment by Juris3D** [comment]
what happens if you map everything in 4.27, and then simply switch project to 5.2? I mean, I make UE assets, I do it in 427. When I test it in UE52, it just reports that it converted everything.

**Comment by JaydenMaalouf** [comment]
> what happens if you map everything in 4.27, and then simply switch project to 5.2? I mean, I make UE assets, I do it in 427. When I test it in UE52, it just reports that it converted everything.

I'm not 100% if it converts your axis mappings

Enhanced Input processes the inputs before it's sent to the input loop, so I would imagine there would be some changes on the plugin side required too

**Comment by luciddream-tsin** [comment]
> what happens if you map everything in 4.27, and then simply switch project to 5.2? I mean, I make UE assets, I do it in 427. When I test it in UE52, it just reports that it converted everything.

### Actually, I can not use this plugin at all in 5.2, as you can see, the plugin recognized my device:


![20230828-185343](https://github.com/JaydenMaalouf/JoystickPlugin/assets/92364028/619b395f-69dd-49fe-86f6-3bba66f5a926)



### But, No matter which button is mapped, it only displays my device name and cannot trigger this event in the blueprint:

![20230828-185352](https://github.com/JaydenMaalouf/JoystickPlugin/assets/92364028/638f502c-e313-4f10-b6e3-5921dad4ad99)

**Comment by VitorAbyssal** [comment]
I am having the same issue, would be great to have Enhanced Input support

**Comment by JaydenMaalouf** [comment]
I'm still working on this. Have been quite busy and am having some issues getting 1:1 setup in my own project. Will hopefully have something in the coming weeks!

**Comment by HeadClot** [comment]
Any word on Enhanced input support?

**Comment by HeadClot** [comment]
Hey @JaydenMaalouf any word on Enhanced Input support?

**Comment by iridescenthsn** [comment]
Hey @JaydenMaalouf great plugin. Is enhanced input support coming? Im kinda relying on this plugin for my project 😅

**Comment by JaydenMaalouf** [comment]
@HeadClot @iridescenthsn you can download the latest pre-release which I've tested and am using in my own project with enhanced input. 

I'm not going to release this as stable until I've tested it in a packaged game but feel free to use it

https://github.com/JaydenMaalouf/JoystickPlugin/releases/tag/3.2.0-rc.10

**Comment by HeadClot** [comment]
Thank you for the update @JaydenMaalouf :)

**Comment by MustangEnthusiast51** [comment]
I'm still getting this issue with the 3.2.0-rc.10 version, is there a fix for it?

**Comment by miju70** [comment]
Hello @JaydenMaalouf, awesome your plugin, it has been really usefull. But now Im working with it on UE5.3 and it seems not to work with EnhancedInputSettings feature. This new feature in UE5.3 changes completely how to remap and save key mappings, but when using that with your plugin, it changes display name on joystickinputselector, but it does not remap the action to the new key. I wonder if its something you can take a look at. :)

**Comment by FredZvt** [comment]
Hi, @JaydenMaalouf! Thanks for your work on the plugin, it's amazing!

I'm trying to use the 3.2.0-rc.10 (JoystickPlugin-5.3.zip) on my project that uses the Enhanced Input System for remapping and the Enhanced Input User Settings for persistence, but it's not working.

My code works perfectly for any keyboard and mouse keys, but if I map one of the joysticks button to the input action, it doesn't trigger the actions nor persists.

I'm using your JoystickInputSelector to get the key and that's working perfectly on the UI. I'm also enabling the `Use Device Name` on the settings and it's showing the names of the devices correctly on the UI, which is great!

Am I missing something?

```
FMapPlayerKeyArgs MapPlayerKeyArgs = FMapPlayerKeyArgs();
MapPlayerKeyArgs.MappingName = TEXT("ToggleLandingGear");	// This needs to be exactly like defined on the input action.
MapPlayerKeyArgs.Slot = EPlayerMappableKeySlot::Unspecified;
MapPlayerKeyArgs.NewKey = SelectedKey.Key;

FGameplayTagContainer FailureReason = FGameplayTagContainer();

EnhancedInputUserSettings->MapPlayerKey(MapPlayerKeyArgs, FailureReason);
EnhancedInputUserSettings->ApplySettings();
EnhancedInputUserSettings->SaveSettings();
```

**Comment by JaydenMaalouf** [comment]
Hey all,
Apologies for the delay - I have been extremely busy with personal life and unfortunately haven't had much time to maintain this (or work on my personal project either ☹️)
BUT! I have finally released [3.3.0](https://github.com/JaydenMaalouf/JoystickPlugin/releases/tag/3.3.0) which contains a fix for #56 which sounds like the same issue as what you are describing

If the issue still persists, can you please raise another issue?
I hope you enjoy the update!
